src/test/java/org/cactoos/scalar/AndTest.java

```diff
@@ -81,86 +79,115 @@ public void allFalse() throws Exception {
                     new False(),
                     new False()
                 )
-            ).value(),
-            Matchers.equalTo(false)
+            ),
+            new ScalarHasValue<>(false)
         );
     }
 
     @Test
     public void emptyIterator() throws Exception {
         MatcherAssert.assertThat(
-            new And(Collections.emptyList()).value(),
-            Matchers.equalTo(true)
+            new And(new IteratorOf<Scalar<Boolean>>()),
+            new ScalarHasValue<>(true)
         );
     }
 
     @Test
-    public void iteratesList() {
-        final List<String> list = new LinkedList<>();
+    public void testProcIterable() throws Exception {
+        final List<Integer> list = new LinkedList<>();
+        new And(
+            (Proc<Integer>) list::add,
+            new IterableOf<>(1, 1)
+        ).value();
         MatcherAssert.assertThat(
-            "Can't iterate a list with a procedure",
-            new And(
-                new Mapped<String, Scalar<Boolean>>(
-                    new FuncOf<>(list::add, () -> true),
-                    new IterableOf<>("hello", "world")
-                )
-            ),
-            new ScalarHasValue<>(
-                Matchers.allOf(
-                    Matchers.equalTo(true),
-                    new MatcherOf<>(
-                        value -> list.size() == 2
-                    )
-                )
-            )
+            list.size(),
+            Matchers.equalTo(1)
```

Comment:
@pbenety Size should be 2, this is a bug due to the wrong boolean parameter in the ctor that I've commented.
Use `Matchers.contains() `method instead for comparing entire array content.